api/v1alpha1/catalog_types.go

```diff
@@ -78,6 +65,32 @@ type GitSource struct {
 	*/
 	// +Optional
 	SecretName *string `json:"secretName,omitempty"`
+
+	// Overrides are the PluginDefinition overrides to be applied
+	// +Optional
+	Overrides []CatalogOverrides `json:"overrides,omitempty"`
+}
+
+type CatalogOverrides struct {
+	// Name is the name of the PluginDefinition to patch with an alias
+	Name string `json:"name"`
+	// Alias is the alias to apply to the PluginDefinition Name via Kustomize patches
+	// For SourceType Helm, this field is passed to postRender Kustomize patch
+	Alias string `json:"alias"`
+	// Repository is the repository to override in the PluginDefinition .spec.helmChart.repository
+	// +Optional
+	Repository string `json:"repository,omitempty"`
+
+	// TODO: implement Options in Overrides for further values patching in PluginDefinition || ClusterPluginDefinition
```

Comment:
tracked the feature in the backlog as well: https://github.com/cloudoperators/greenhouse/issues/1557